hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -102,6 +102,18 @@ public HoodieFileGroupReader(HoodieReaderContext<T> readerContext,
     this.start = 0;
     this.length = Long.MAX_VALUE;
     this.baseFileIterator = new EmptyIterator<>();
+    this.shouldUseRecordPosition = false;
```

Comment:
shouldUseRecordPosition is always false now, do we still need to keep it.

Reply:
Oh, I see, there is no need to keep 2 constructors here, we can instantiate the record buffer from it's factory class with the given flag `shouldUseRecordPosition`

Reply:
Remove the shouldUseRecordPosition variable.